### DISCUSSION TITLE
Make EOS packable by Python/PyPI

### DISCUSSION THREAD

**Discussion by b0661** [Akkudoktor-EOS/EOS#106] [Ideas]
I'm interested in using EOS as a library and application to be provided by [PyPI](https://pypi.org/) 

The current directory structure does not support that easily. There are some best practice directory layouts suggested in [the Python packaging tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/). Whereas a mapping can always be configured, it is more easy and less error prone if the directory structure makes it quite clear which files will be part of a Python package.

EOS as a package name is already in use in PyPI. The package name has to be unique in [PyPI](https://pypi.org/) . Ideally the package name and the direcory name are the same.

My suggestion is to create a directory structure for the packages and move the existing files therein.

├── LICENSE
├── pyproject.toml
├── README.md
├── src/
│   └── Akkudoktor_EOS/ ------------------ the library (mostly from current modules directory)
│       ├── __init__.py
│       ├── class_akku.py
│       └── ....py
│   └── Akkudoktor_EOS_Server/ -------- the flask server
│       ├── __init__.py
│       └── flask_server.py
└── tests/
└── ...       as now, besides modules/

Maybe a short for Akkudoktor_EOS such as AEOS may be convenient.

Another benefit of packaging is that you can install the package(s) in development mode (basically a link to the source directory) and use them as an installed package (no Python PATH hassle). 

Cheers
Bobby

**Comment by michaelosthege** [comment]
I agree that a standard Python package structure, and making it `pip`-installable would help.
We'd have to rename the `modules` directory to align with the package name, add some things to `pyproject.toml` and update the imports.

@drbacke would you have a preferred name for the package?
https://pypi.org/search/?q=EOS shows that `eos` and `python-eos` are taken, but `aeos`, `eos-core`, `akkudoktor-eos`, `eos-server`, for example are free.

**Reply by drbacke** [reply]
I also agree: this will help. 
akkudoktor-eos maybe helpful for future promotion

**Reply by michaelosthege** [reply]
ok, shall I make a PR?

It helps that there are no other PRs open at the moment.

**Reply by b0661** [reply]
I have a PR available. Just have to adapt the naming.

**Comment by b0661** [comment]
This is now implemented and merged.

**Reply by drbacke** [reply]
Works quite well, but there is a new small bug. 
https://github.com/Akkudoktor-EOS/EOS/issues/121
